src/custom/components/Stepper/index.tsx

```diff
@@ -0,0 +1,128 @@
+import styled from 'styled-components/macro'
+import CheckCircle from 'assets/cow-swap/check.svg'
+import { transparentize } from 'polished'
+
+export const Wrapper = styled.div`
+  width: 100%;
+  display: flex;
+  flex-flow: row wrap;
+  margin: 12px 0 24px;
+`
+
+export const Step = styled.div<{
+  totalSteps: number
+  isActiveStep: boolean
+  completedStep: boolean
+  circleSize?: number
+}>`
+  display: flex;
+  flex-flow: column wrap;
+  align-items: center;
+  position: relative;
+  flex: 1 1 ${({ totalSteps }) => `calc(100% / ${totalSteps})`};
+
+  &::before,
+  &::after {
+    content: '';
+    position: absolute;
+    top: ${({ circleSize }) => (circleSize ? `calc(${circleSize / 2}px` : '21px')};
+    height: 1px;
+    border-top: 1px solid ${({ theme }) => theme.border2};
+  }
+
+  &::before {
+    left: 0;
+    right: 50%;
+    margin-right: ${({ circleSize }) => (circleSize ? `${circleSize}px` : '42px')};
```

Comment:
could be useful to cache `42px` here as you're clearly setting it in several places. would make tweaking this a breeze later and prevent 1 change from skewing the others

Reply:
Agreed. Will set up CSS vars in the parent.

Reply:
@W3stside Updated to reflect this now.